activerecord/lib/active_record/connection_adapters/abstract_adapter.rb

```diff
@@ -636,7 +636,15 @@ def disable_referential_integrity
 
       # Override to check all foreign key constraints in a database.
       def all_foreign_keys_valid?
+        check_all_foreign_keys_valid!
         true
+      rescue StandardError
```

Comment:
this `rescue Standard` error look a bit fishy, as it's pretty much a "rescue all". Is there no base exception expected?

Reply:
Ah it's for the `raise String` here: https://github.com/rails/rails/pull/47990/files#diff-42d7246995482d3a13c58da637885c7a85c9dbcb66e65f8460d14e65f7462ea9R664

Then it's `RuntimeError`.

Reply:
Oops, no, it should be a `ActiveRecord::StatementInvalid`. Fixed.